packages/backend/src/models/CommentModel/CommentModel.ts

```diff
@@ -0,0 +1,229 @@
+import { Comment, LightdashUser, NotFoundError } from '@lightdash/common';
+import { Knex } from 'knex';
+import {
+    DashboardsTableName,
+    DashboardTileCommentsTableName,
+    DashboardTilesTableName,
+    DashboardVersionsTableName,
+    DbDashboardTileComments,
+} from '../../database/entities/dashboards';
+import { UserTableName } from '../../database/entities/users';
+
+type CommentModelDependencies = {
+    database: Knex;
+};
+
+export class CommentModel {
+    private readonly database: Knex;
+
+    constructor(deps: CommentModelDependencies) {
+        this.database = deps.database;
+    }
+
+    private async checkDashboardTileExistsInDashboard(
+        dashboardUuid: string,
+        dashboardTileUuid: string,
+    ) {
+        // NOTE: ensure that this dashboard actually contains the tile, since tile uuids might not be unique across dashboards
+        const dashboardTile = await this.database(DashboardTilesTableName)
+            .join(
+                DashboardVersionsTableName,
+                `${DashboardVersionsTableName}.dashboard_version_id`,
+                '=',
+                `${DashboardTilesTableName}.dashboard_version_id`,
+            )
+            .join(
+                DashboardsTableName,
+                `${DashboardsTableName}.dashboard_id`,
+                '=',
+                `${DashboardVersionsTableName}.dashboard_id`,
+            )
+            .leftJoin(
+                'dashboard_tile_charts',
+                'dashboard_tile_charts.dashboard_tile_uuid',
+                'dashboard_tiles.dashboard_tile_uuid',
+            )
+            .leftJoin(
+                'saved_queries',
+                'saved_queries.saved_query_id',
+                'dashboard_tile_charts.saved_chart_id',
+            )
+            .where(`${DashboardsTableName}.dashboard_uuid`, dashboardUuid)
+            .andWhere(
+                `${DashboardTilesTableName}.dashboard_tile_uuid`,
+                dashboardTileUuid,
+            )
+            .select(
+                `${DashboardTilesTableName}.dashboard_tile_uuid`,
+                `saved_queries.saved_query_uuid`,
+            )
+            .first();
+
+        if (!dashboardTile) {
+            throw new NotFoundError('Dashboard tile not found for dashboard');
+        }
+        return { savedChartUuid: dashboardTile.saved_query_uuid };
```

Comment:
this will also work for `dashboarTile`s that have no saved_query_uuid right? I think 🤔 So if they are Looms or Markdowns

Reply:
yes, I tested that, they return null, so we insert nothing on `saved_chart_uuid`

![image](https://github.com/lightdash/lightdash/assets/1983672/3c802213-ebc8-48ce-8f0f-7cd1ec7fa028)